velocity_smoother:
  ros__parameters:
    input_cmd_vel_topic: "/cmd_vel_raw"
    output_cmd_vel_topic: "/cmd_vel_smoothed"
    smoothing_frequency: 20.0
    scale_velocities: False
    feedback: "OPEN_LOOP"
    max_velocity: [0.6, 0.0, 0.6]
    min_velocity: [-0.6, 0.0, -0.6]
    max_accel: [1.1, 0.0, 1.1]  # [2.5, 0.0, 3.2]
    max_decel: [-1.1, 0.0, -1.1] # [-2.5, 0.0, -3.2]
    odom_topic: "odom"
    odom_duration: 0.1
    deadband_velocity: [0.05, 0.0, 0.05]
    velocity_timeout: 1.0
    enable_stamped_cmd_vel: false

collision_monitor:
  ros__parameters:
    base_frame_id: "base_footprint"
    odom_frame_id: "odom"
    cmd_vel_in_topic: "/cmd_vel_smoothed"
    cmd_vel_out_topic: "/cmd_vel"
    state_topic: "collision_monitor_state"
    transform_tolerance: 0.5
    source_timeout: 1.0
    base_shift_correction: true
    stop_pub_timeout: 2.0
    # Polygons represent zone around the robot for "stop", "slowdown" and "limit" action types,
    # and robot footprint for "approach" action type.
    polygons: ["PolygonStop", "PolygonSlow"]
    PolygonStop: 
      type: "circle"
      radius: 0.49
      action_type: "stop"
      min_points: 5
      visualize: true
      enabled: true
      stop_pub_timeout: 0.5
    PolygonSlow:
      type: "circle"
      radius: 0.78
      action_type: "slowdown"
      min_points: 4
      slowdown_ratio: 0.5
      visualize: true
      enabled: true
    observation_sources: ["merged_slow"]
    merged_slow:
      type: scan
      topic: /scan
      min_range: 0.15
      max_range: 10.0
      enabled: true

